### ISSUE TITLE
Improve Sequence Run Info

### ISSUE THREAD

**Issue by victorskl** [umccr/data-portal-apis#239]
Context:

- There are desirable use cases to ingest `RunInfo.xml` into Portal DB; tentatively to `SequenceRun` table

Examples:
- Such as we could infer/reconstruct Override Cycle at runtime; instead of error prone manual data entry in Metadata sheet
- Few cases in when we drive Workflow launch

Pointer:
- Tentatively, we could ingest such that the first time when we received BSSH Run event, fetch `RunInfo.xml` from GDS -- code entrypoint could be around [handle_bssh_run_event](https://github.com/umccr/data-portal-apis/blob/dev/data_processors/pipeline/lambdas/sqs_iap_event.py#L76) 
- Transform appropriate format e.g. as plain TextField or JSONField or impl-as-see-fit
- Follow up by few functions impl that infer Override Cycle and/or impl-as-see-fit -- code could live in `liborca` or under `domain` package or `sequence_srv` module or impl-as-see-fit

Actions:
- @alexiswl and @reisingerf will work on it

**Comment by victorskl** [comment]
This may be indirectly related (or potentially fix) to #231 and/or #229

**Comment by reisingerf** [comment]
Notes to self:
Uh, not as straight forward as I thought, but doable.

I thought we could just add the RunInfo to the `SequenceRun` object as soon as that's created. I forgot that this is not a representative of a run, but rather a run event, i.e. multiple `SequenceRun` objects are generated for the same run (depending on `status` and `date_modified`).
To work around this we could attach that RunInfo information to a specific object (unique `run_id` + `date_modified` + `status`), or propagate it across all objects of the same run (retrieving the RunInfo.xml only once from GDS).
(multiple SequenceRun objects for the same run makes is also a little more difficult to map libraries to sequence runs. One to keep in mind for the domain model)

Also, the `bcl_convert.handler` (that would use that info to double check/select the OverrideCycles) does not require a `SequenceRun` object (to support manual invocations where no `SequenceRun` DB entries exist), so we can't rely on that way to retrieve RunInfo data. We could skip the RunInfo check if that info is not available (which is not needed for the regular case anyway) or require it being passed into the handler.

**Comment by alexiswl** [comment]
Is it worth having a static run object that represents a run, contains the run id, gds path to the run, run name (with flowcell index), run number (to match metadata) and then the cycles info? SequenceRuns could be renamed to sequence run events that map to the run id? Library IDs could also then map to the static run object.

**Comment by reisingerf** [comment]
Yes, that's the idea, separate the event/execution states from the actual domain entities. 
But I think we'll leave the remodelling for the new EventBus version.
